mantidproject/mantid
# Error Reporter Save name and email locally

**Description of work.**

- User ContactInfo can now be saved and prefilled in the errorreporter to increase the number of reports with contact info (more convenient for users)
- It is now possible to send a textbox when selecting Share Non-Identifiable info. This also negates the need for fixing when to grey/ungrey out this button based on the textbox. This button is now only greyed out if the name or email are altered from empty or the saved+prefilled name and email.
- I tried to fix the tab ordering but QT seems to want to include other elements despite me setting their FocusPolicy to ClickFocus. I don't think this is important as with the initial focus being on the textbox, tabbing here just adds a tab to the text rather than jumping between elements.
- 
**To test:**

Either 1. setup a local error reporter server using the readme here: https://github.com/mantidproject/errorreports
(basically pull that repo, add the .env file, install docker-compose, run sudo bin/boot.sh and make sure to add `errorreports.rooturl = http://localhost:8082` to your Mantid.user.properties file)
or 2. be happy with spamming the ErrorReporter Slack channel for a bit!

- Make sure you have downloaded the TrainingCourseData
- In MantidWorkbench, Load `TrainingCourseData/Training_Exercise3a_SNS.nxs`
- Then open the algorithm dialog for NormaliseToMonitor
-  you should get an error reporter appear, but Mantid stays open, so you can open the error reporter many times quickly.

- Open the error reporter
- Input your name and email, tick the new SaveContactInfo checkbox and click `Share All Info`
- Open the error reporter: have the name and email prefilled and is the checkbox preticked?
- Try again with just the email and just the name. Notice how any changes get updated as long as the new checkbox is ticked
- Open the error reporter and Uncheck the tickbox
- Reopen and notice that the ContactInfo was deleted (the contact_info file was emptied)

- Also notice that you can type in the textbox and send non-identifiable info aslong as the name and email are not altered.

Fixes https://github.com/mantidproject/mantid/issues/28083



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
